photon-api/src/integTest/scala/com/linkedin/photon/ml/model/GAMEModelTest.scala

```diff
@@ -279,7 +277,6 @@ class GAMEModelTest extends SparkTestUtils {
     val RE2Model = new RandomEffectModel(glmRE2RDD, "REModel2", "RE2Features")
 
     // This Game model has 1 fixed effect, and 2 different random effect models
-    GAMEModel(LOGISTIC_REGRESSION, ("fixed", FEModel), ("RE1", RE1Model), ("RE2", RE2Model))
-      .checkInvariants() // this should throw: we have a stray PoissonRegressionModel
```

Comment:
In favor of having variables that indicate the type in each model, and a check for that? The essence of the `checkInvariants`, that all submodels need to be of the same type, still has to be verified somewhere.